src/api/api.ts

```diff
@@ -0,0 +1,31 @@
+import type { Comments, Product, Products } from "./apiTypes";
+
+const BASE_URL = "https://panda-market-api.vercel.app";
+
+export async function getProducts({ page=1, pageSize=10, orderBy="recent", keyword='' }: {
+    page?: number;
+    pageSize?: number;
+    orderBy?: "recent" | "favorite";
+    keyword?: string;
+}): Promise<Products> {
+    const response = await fetch(`${BASE_URL}/products?page=${page}&pageSize=${pageSize}&orderBy=${orderBy}${keyword && "&keyword="+keyword}`);
+    if (!response.ok) { throw Error("Request Error"); }
+    return (await response.json()) as Products;
+}
```

Comment:
💬 여담
keyword는 옵셔널 파람인데 기본값으로 빈 문자열을 줄 필요가 있을까요?
위의 코드도 동작상 문제가 없을 것 같지만 가독성 측면에서 기본값이 없는것이 더 좋을 것 같아요~

```suggestion
export async function getProducts({ page=1, pageSize=10, orderBy="recent", keyword }: {
    page?: number;
    pageSize?: number;
    orderBy?: "recent" | "favorite";
    keyword?: string;
}): Promise<Products> {
    const response = await fetch(`${BASE_URL}/products?page=${page}&pageSize=${pageSize}&orderBy=${orderBy}${keyword && "&keyword="+keyword}`);
    if (!response.ok) { throw Error("Request Error"); }
    return (await response.json()) as Products;
}
```

Reply:
기본값을 주지 않으면 query parameter에 그대로 undefined가 뒤에 붙어서 기본값을 줬습니다~